gator/fragments.py

```diff
@@ -1,64 +1,65 @@
-        return False
-    # if the paragraph is a fenced code block, it is not a paragraph
-    if candidate.startswith(CODE_FENCE_MARKER):
-        return False
-    # there may be other edge cases that should be added here in the
-    # future -- what other structures look like paragraphs but should
-    # not be?
-    # if nothing has returned by now, the candidate must be a paragraph
-    return True
+def is_single_subnode(subnode):
+    """Determines if a node is single, and doesn't have a closing tag"""
+    return (
+        subnode.t == "text"
+        or subnode.t == "thematic_break"
+        or subnode.t == "html_block"
+        or subnode.t == "code_block"
+        or subnode.t == "code"
+        or subnode.t == "softbreak"
+        or subnode.t == "linebreak"
+    )
 
 
-def get_paragraphs(contents, blank_replace=True):
+def get_paragraphs(contents):
     """Retrieves the paragraphs in the writing"""
-    # use a replacement to handle a string with just spaces
-    if blank_replace is True:
-        contents = contents.replace(SPACE, NOTHING)
-    # replace a single newline with a blank space, respecting double newlines
-    contents = contents.replace(DOUBLE_NEWLINE, GATORGRADER_REPLACEMENT)
-    contents = contents.replace(NEWLINE, SPACE)
-    contents = contents.replace(GATORGRADER_REPLACEMENT, DOUBLE_NEWLINE)
-    pattern = re.compile(PARAGRAH_RE)
-    paragraphs = pattern.findall(contents)
-    # disregard all of the section headers in markdown
-    matching_paragraphs = []
-    # iterate through all potential paragraphs and gather
-    # those that match the standard for legitimacy
-    for paragraph in paragraphs:
-        if is_paragraph(paragraph) is True:
-            matching_paragraphs.append(paragraph)
-    return matching_paragraphs
+    ast = commonmark.Parser().parse(contents)
+    mode_looking = True
+    paragraph_list = []
+    paragraph_content = ""
+    counter = 0
+    # Iterates through the file to find the paragraphs and add them to the array
+    for subnode, enter in ast.walker():
+        if mode_looking:
+            # Checks to see if it's at the opening of a paragraph subnode
+            if counter == 1 and subnode.t == "paragraph" and enter:
+                # Sets up paragraph content to go to the array
+                paragraph_content = ""
+                # Stops search for paragraph nodes, as one has been found
+                mode_looking = False
+        else:
+            # Checks to see if it has found the closing of the paragraph subnode
+            if counter == 2 and subnode.t == "paragraph" and not enter:
+                # Adds the content of the paragraph to the array
+                paragraph_list.append(paragraph_content)
+                # Starts a search for a new paragraph
+                mode_looking = True
+            if subnode.literal is not None:
+                paragraph_content += subnode.literal
+
+        # Looks for an opening and a closing node
```

Comment:
This documentation comment is misleading -- this code does not look for an opening and closing node, it keeps track of the depth inside the tree we are currently at.